### PR TITLE
add reward actor

### PR DESCRIPTION
**Pull request by zixuanzh**
**In this PR**

- Add reward actor that handles
    - Block reward minting (minting function left as TODO)
    - Token vesting (genesis and block reward)

### CODE REVIEW

**Comment by mishmosh** [comment]
@anorth is out tomorrow, tagging @jbenet for review instead.

File: src/systems/filecoin_vm/sysactors/_index.md [hunk 1]
```diff
@@ -5,10 +5,13 @@ entries:
...
+- reward_actor
 ---
 
 - There are two system actors required for VM processing:
   - [CronActor](#CronActor) - runs critical functions at every epoch
   - [InitActor](#InitActor) - initializes new actors
-- There is one more VM level actor:
+- There are two more VM level actors:
   - [AccountActor](#AccountActor) - for user accounts.
+  - [RewardActor](#RewardActor) - for block reward and token vesting
```
Comment (jzimmerman) [inline src/systems/filecoin_vm/sysactors/_index.md]: nit: may be worth clarifying here that RewardActor is a system/singleton (but AccountActor is not)

File: src/systems/filecoin_vm/sysactors/reward_actor.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		rt.AbortStateMsg("ra._withdrawReward: ownerAddr not found in RewardMap.")
+	}
+
+	rewardToWithdraw := actor.TokenAmount(0)
+	indicesToRemove := make([]int, len(rewards))
+
+	for i, r := range rewards {
+		elapsedEpoch := rt.CurrEpoch() - r.StartEpoch()
+		unlockedReward := actor.TokenAmount(uint64(r.ReleaseRate()) * uint64(elapsedEpoch))
+		withdrawableReward := unlockedReward - r.WithdrawAmount()
```
Comment (jzimmerman) [inline src/systems/filecoin_vm/sysactors/reward_actor.go]: [Blocking] Assert nonnegative

File: src/systems/filecoin_vm/sysactors/reward_actor.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+////////////////////////////////////////////////////////////////////////////////
+
+func (st *RewardActorState_I) _withdrawReward(rt vmr.Runtime, ownerAddr addr.Address) actor.TokenAmount {
+
+	rewards, found := st.RewardMap()[ownerAddr]
+	if !found {
+		rt.AbortStateMsg("ra._withdrawReward: ownerAddr not found in RewardMap.")
+	}
+
+	rewardToWithdraw := actor.TokenAmount(0)
```
Comment (jzimmerman) [inline src/systems/filecoin_vm/sysactors/reward_actor.go]: nit: rename to `rewardToWithdrawTotal`?

File: src/systems/filecoin_vm/sysactors/reward_actor.id [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import vmr "github.com/filecoin-project/specs/systems/filecoin_vm/runtime"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+
+type Reward struct {
+    StartEpoch      block.ChainEpoch
+    Value           actor.TokenAmount
+    ReleaseRate     actor.TokenAmount  // linear slope, unit: FIL per Epoch
+    WithdrawAmount  actor.TokenAmount
```
Comment (jzimmerman) [inline src/systems/filecoin_vm/sysactors/reward_actor.id]: [Blocking] rename to `AmountWithdrawn` (or something similar) to avoid confusion

**Pull request merged**